client/rest-high-level/src/main/java/org/elasticsearch/client/security/user/privileges/Role.java

```diff
@@ -0,0 +1,285 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.client.security.user.privileges;
+
+import org.elasticsearch.common.Nullable;
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.xcontent.ConstructingObjectParser;
+import org.elasticsearch.common.xcontent.ToXContentObject;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentHelper;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.common.xcontent.XContentType;
+import org.elasticsearch.common.xcontent.ObjectParser.ValueType;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+import static org.elasticsearch.common.xcontent.ConstructingObjectParser.constructorArg;
+import static org.elasticsearch.common.xcontent.ConstructingObjectParser.optionalConstructorArg;
+
+/**
+ * Represents an aggregation of privileges. This does not have a name
+ * identifier.
+ */
+public final class Role implements ToXContentObject {
+
+    public static final String NONE_CLUSTER_PRIVILEGE_NAME = "none";
+    public static final String ALL_CLUSTER_PRIVILEGE_NAME = "all";
+    public static final String MONITOR_CLUSTER_PRIVILEGE_NAME = "monitor";
+    public static final String MONITOR_ML_CLUSTER_PRIVILEGE_NAME = "monitor_ml";
+    public static final String MONITOR_WATCHER_CLUSTER_PRIVILEGE_NAME = "monitor_watcher";
+    public static final String MONITOR_ROLLUP_CLUSTER_PRIVILEGE_NAME = "monitor_rollup";
+    public static final String MANAGE_CLUSTER_PRIVILEGE_NAME = "manage";
+    public static final String MANAGE_ML_CLUSTER_PRIVILEGE_NAME = "manage_ml";
+    public static final String MANAGE_WATCHER_CLUSTER_PRIVILEGE_NAME = "manage_watcher";
+    public static final String MANAGE_ROLLUP_CLUSTER_PRIVILEGE_NAME = "manage_rollup";
+    public static final String MANAGE_INDEX_TEMPLATES_CLUSTER_PRIVILEGE_NAME = "manage_index_templates";
+    public static final String MANAGE_INGEST_PIPELINES_CLUSTER_PRIVILEGE_NAME = "manage_ingest_pipelines";
+    public static final String TRANSPORT_CLIENT_CLUSTER_PRIVILEGE_NAME = "transport_client";
+    public static final String MANAGE_SECURITY_CLUSTER_PRIVILEGE_NAME = "manage_security";
+    public static final String MANAGE_SAML_CLUSTER_PRIVILEGE_NAME = "manage_saml";
+    public static final String MANAGE_PIPELINE_CLUSTER_PRIVILEGE_NAME = "manage_pipeline";
+    public static final String MANAGE_CCR_CLUSTER_PRIVILEGE_NAME = "manage_ccr";
+    public static final String READ_CCR_CLUSTER_PRIVILEGE_NAME = "read_ccr";
```

Comment:
I would move these to a separate class, just to keep the size of this class down, but it's up to you.

Reply:
👍